component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2690,7 +2767,97 @@ private boolean isEmailUpdateFailed(AuthenticationContext context) {
+
+        return Boolean.TRUE.equals(context.getProperty(IS_IDENTIFIER_FIRST_INITIATED_FROM_AUTHENTICATOR));
+    }
+
+    private String validateIdentifierFromRequest(HttpServletRequest request) throws AuthenticationFailedException {
+
+        String identifierFromRequest = request.getParameter(USER_NAME);
+        if (StringUtils.isBlank(identifierFromRequest)) {
+            throw new InvalidCredentialsException(EmailOTPAuthErrorConstants.ErrorMessages.EMPTY_USERNAME.getCode(),
+                    EmailOTPAuthErrorConstants.ErrorMessages.EMPTY_USERNAME.getMessage());
+        }
+        return identifierFromRequest;
+    }
+
+    /**
+     * This method is used to resolve the user from authentication response from identifier handler.
+     *
+     * @param request The httpServletRequest.
+     * @param context The authentication context.
+     * @throws AuthenticationFailedException In occasions of failing.
+     */
+    public org.wso2.carbon.user.core.common.User resolveUser(HttpServletRequest request, AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        String username = validateIdentifierFromRequest(request); // obtains the username from the request
+        username = FrameworkUtils.preprocessUsername(username, context);
+        org.wso2.carbon.user.core.common.User user = new org.wso2.carbon.user.core.common.User();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        String tenantDomain = MultitenantUtils.getTenantDomain(username);
+
+        user.setUsername(tenantAwareUsername);
+        user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(username));
+        user.setTenantDomain(tenantDomain);
+        return user;
+    }
+
+    /**
+     * This method is used to set the resolved user in context.
+     *
+     * @param context The authentication context.
+     * @param user    The user object
+     */
+    private void setResolvedUserInContext(AuthenticationContext context, org.wso2.carbon.user.core.common.User user) {
+
+        setSubjectInContext(context, user);
+        context.setProperty("user", user);
+    }
+
+    /**
+     * This method is used to set the subject in context.
+     *
+     * @param context The authentication context
+     * @param user    The user object
+     */
+    private void setSubjectInContext(AuthenticationContext context, org.wso2.carbon.user.core.common.User user) {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser(user);
+        context.setSubject(authenticatedUser);
+    }
+
+    /**
+     * This method is used to set the resolved user in context.
+     *
+     * @param context  The authentication context
+     * @param request  Request
+     * @param response Response
+     * @throws AuthenticationFailedException AuthenticationFailedException
+     */
+    private void redirectUserToIdf(HttpServletRequest request, HttpServletResponse response,
```

Comment:
what is the meaning of redirectUserToIdf?

Reply:
It is the method which redirects user to the Identity-First authentication page.